crates/ruff_linter/src/rules/flake8_markupsafe/rules/unsafe_markup_use.rs

```diff
@@ -0,0 +1,100 @@
+use ruff_python_ast::ExprCall;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::name::QualifiedName;
+use ruff_text_size::Ranged;
+
+use crate::{checkers::ast::Checker, settings::LinterSettings};
+
+/// ## What it does
+/// Checks for non-literal strings being passed to [`markupsafe.Markup`].
+///
+/// ## Why is this bad?
+/// `markupsafe.Markup` does not perform any escaping, so passing dynamic
+/// content, like f-strings, variables or interpolated strings will potentially
+/// lead to XSS vulnerabilities.
+///
+/// Instead you should interpolate the [`markupsafe.Markup`] object.
+///
+/// Using [`lint.flake8-markupsafe.extend-markup-names`] additional objects
+/// can be treated like [`markupsafe.Markup`].
+///
+/// ## Example
+/// ```python
+/// from markupsafe import Markup
+///
+/// content = "<script>alert('Hello, world!')</script>"
+/// html = Markup(f"<b>{content}</b>")  # XSS
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from markupsafe import Markup
+///
+/// content = "<script>alert('Hello, world!')</script>"
+/// html = Markup("<b>{}</b>").format(content)  # Safe
+/// ```
+///
```

Comment:
Consider adding the `## Options` section here, example: https://docs.astral.sh/ruff/rules/ambiguous-unicode-character-comment/#options